src/tshwctl.c

```diff
@@ -26,26 +27,63 @@ void do_info(void)
 {
 	fpga_init(0x50004000);
 	eval_cmd_init();
-
+	
 	printf("MODEL=%X\n", model);
 
 	if(model == 0x7100) {
 		printf("FPGA_REV=0x%X\n", fpeek32(0x0) >> 16);
 		printf("CPU_OPTS=0x%X\n", eval_cmd("cpu_opts"));
 		printf("IO_OPTS=0x%X\n", eval_cmd("io_opts"));
 		printf("IO_MODEL=0x%X\n", eval_cmd("io_model"));
+
 	} else if(model == 0x7250) {
 		uint32_t fpga_rev = fpeek32(0x0);
 		uint32_t fpga_hash = fpeek32(0x4);
+		uint8_t reg = (fpeek32(0x8) & 0xF); 
+		uint8_t straps = 0;
+		const uint8_t avail_straps = 0x7; /* Current available strapping (MSB)R229, R227, R228(LSB) */
+
+		
+		/* 
+		* Shift bits around to read straps correctly
+		* FPGA implementation differs from schematic
+		* e.g. R229,R227,R228,R230 (fpga) vs. R230,R229,R227,R228 (.sch) 
+		* where R230 is the designated strapping for RAM
+		*/
+		
+		straps |= (!!(reg & (1<<1))); 		/* R228 TXD2_485*/ 
+		straps |= (!!(reg & (1<<2))) << 1;  /* R227 COM3_TXD*/
+		straps |= (!!(reg & (1<<3))) << 2;  /* R229 COM2_TXD*/
+		straps ^= avail_straps;	
+
 		printf("FPGA_REV=%d\n", fpga_rev & 0x7fffffff);
+		printf("OPTS=0x%X\n", straps);
 
 		if(fpga_rev & (1 << 31))
 			printf("FPGA_HASH=\"%x-dirty\"\n", fpga_hash);
 		else
 			printf("FPGA_HASH=\"%x\"\n", fpga_hash);
 
-		printf("OPTS=0x%X\n", fpeek32(0x8) & 0xF);
```

Comment:
Since the previous style just printed the value, is this preferable to printing the whole model?

Reply:
I would print both to allow this to be tracked for future variants.  The full part number is a "nice to have" that will cover most products for users to be able to look up their boards, but we don't always change this util for something like a custom population.